custom/dfci_swasth/rules/custom_actions.py

```diff
@@ -0,0 +1,81 @@
+from corehq.apps.data_interfaces.models import CaseRuleActionResult
+from corehq.apps.hqcase.utils import update_case, AUTO_UPDATE_XMLNS
+from corehq.form_processor.exceptions import CaseNotFound
+from corehq.form_processor.models import CommCareCase
+from custom.dfci_swasth.constants import (
+    CASE_TYPE_PATIENT,
+    PROP_CCUSER_CASELOAD_CASE_ID,
+    PROP_COUNSELLOR_CLOSED_CASE_LOAD,
+    PROP_COUNSELLOR_LOAD,
+)
+
+
+def update_counsellor_load(patient_case, rule):
+    num_related_updates = 0
+
+    if patient_case.type != CASE_TYPE_PATIENT:
+        return CaseRuleActionResult(num_related_updates=num_related_updates)
+
+    ccuser_caseload_case = _get_ccuser_caseload_case(patient_case)
+
+    # Updating the case type 'ccuser_caseload' case properties
+    if ccuser_caseload_case:
+        case_updates = _get_case_updates(ccuser_caseload_case)
+        if case_updates:
+            (submission, cases) = update_case(
+                ccuser_caseload_case.domain,
+                ccuser_caseload_case.case_id,
+                case_properties=case_updates,
+                xmlns=AUTO_UPDATE_XMLNS,
+                device_id=__name__ + ".update_counsellor_load",
+                form_name=rule.name,
+            )
+            num_related_updates = 1
+            rule.log_submission(submission.form_id)
+
+    return CaseRuleActionResult(
+        num_related_updates=num_related_updates,
+    )
+
+
+def _get_ccuser_caseload_case(patient_case):
+    coun_ccuser_caseload_case_id = patient_case.get_case_property(PROP_CCUSER_CASELOAD_CASE_ID)
+    if coun_ccuser_caseload_case_id:
+        try:
+            return CommCareCase.objects.get_case(coun_ccuser_caseload_case_id, domain=patient_case.domain)
+        except CaseNotFound:
+            return None
+
+
+def _get_case_updates(ccuser_caseload_case):
+    counsellor_load = _get_updated_counsellor_load(ccuser_caseload_case)
+    counsellor_closed_case_load = _get_updated_counsellor_closed_case_load(ccuser_caseload_case)
+
+    result = {}
+    if counsellor_load is not None:
+        result.update({PROP_COUNSELLOR_LOAD: counsellor_load})
+    if counsellor_closed_case_load is not None:
+        result.update({PROP_COUNSELLOR_CLOSED_CASE_LOAD: counsellor_closed_case_load})
+    return result
+
+
+def _get_updated_counsellor_load(ccuser_caseload_case):
+    counsellor_load = _get_integer_case_property_value(ccuser_caseload_case, PROP_COUNSELLOR_LOAD)
+    # Value of counsellor load can be minimum 0
+    if counsellor_load is not None and counsellor_load >= 1:
```

Comment:
Added explicit `None` check here as value of `counsellor_load` can be zero as well. In that case, `if counsellor_load` check will evaluate to false which is wrong for this case.

Reply:
The [below line](https://github.com/dimagi/commcare-hq/pull/33364/files#r1314645906) is okay since you are 'trueing' on zero (`>=0`) on 2nd condition, but here you are 'trueing' only on `>= 1`, which means this condition could simply be `if counsellor_load`.